unstructured_ingest/v2/processes/connectors/onedrive.py

```diff
@@ -348,27 +352,19 @@ def run(self, path: Path, file_data: FileData, **kwargs: Any) -> None:
             # Attempt to get the folder
             folder = drive.root.get_by_path(destination_folder_str)
             folder.get().execute_query()
-        except Exception:
+        except ClientRequestException as e:
             # Folder doesn't exist, create it recursively
-            current_folder = drive.root
-            for part in destination_folder.parts:
-                # Use filter to find the folder by name
-                folders = (
-                    current_folder.children.filter(f"name eq '{part}' and folder ne null")
-                    .get()
-                    .execute_query()
-                )
-                if folders:
-                    current_folder = folders[0]
-                else:
-                    # Folder doesn't exist, create it
-                    current_folder = current_folder.create_folder(part).execute_query()
-            folder = current_folder
+            root = drive.root
+            root_folder = self.upload_config.root_folder
+            if e.message != "The resource could not be found.":
```

Comment:
I wonder, is there a better way to know if it's an error? Not ideal to check a string like this.